butterfly-slack/src/main/java/com/paypal/butterfly/slack/SlackNotifierConfig.java

```diff
@@ -0,0 +1,121 @@
+            @Override
+            public void postTransformation(TransformationRequest transformationRequest, TransformationResult transformationResult) {
+                postNotification(transformationResult, "postSuccess.json");
+            }
+
+            @Override
+            public void postTransformationAbort(TransformationRequest transformationRequest, TransformationResult transformationResult) {
+                postNotification(transformationResult, "postAbort.json");
+            }
+
+            private void postNotification(TransformationResult transformationResult, String jsonFile){
+                String jsonPath = new File("").getAbsolutePath().concat("/butterfly-slack/src/main/resources/" + jsonFile);
+                String token = System.getProperty("slack.token");
+                try {
+                    String jsonFileString = new String(Files.readAllBytes(Paths.get(jsonPath)));
+                    Client client = ClientBuilder.newClient();
+                    WebTarget target = client.target("https://www.slack.com/api/");
+                    String[] butterflyUserIDs = transformationResult.getUserId().split(" ");
+                    butterflyUserIDs = new String[]{"edwardhwang@berkeley.edu", "ehwang0824@gmail.com"};
+                    String requestID = transformationResult.getId();
+                    for (String id : butterflyUserIDs) {
+                        //TODO: Change hard coded email into butterflyUserIDs
+                        String email = id + ""; //id + "@paypal.com";
+                        String slackUserID = getSlackUserID(target, token, email);
+                        String slackChannelID = getSlackChannelID(target, token, slackUserID);
+                        postSlackMessage(target, token, slackChannelID, jsonFileString, requestID);
+                    }
+                } catch (IOException e) {
+                    logger.warn("No JSON message found for slack notification");
+                    e.printStackTrace();
+                }
+            }
+
+            /**
+             * GET slack user IDs from PayPal email using the users.lookupByEmail web api call
+             */
+            private String getSlackUserID(WebTarget target, String token, String email){
+                    JSONObject slackResponse = new JSONObject(target
+                            .path("users.lookupByEmail")
+                            .queryParam("token", token)
+                            .queryParam("email", email)
+                            .request(MediaType.APPLICATION_JSON)
+                            .get(String.class));
+                    String slackUserID = "";
+                    try {
+                        slackUserID = slackResponse.getJSONObject("user").getString("id");
+                    } catch (Exception e) {
+                        logger.warn("An exception happened when retrieving the username for slack " + slackResponse.getString("error") + " " + e);
```

Comment:
Never concatenate an exception in a log statement. Add it as a second parameter instead, otherwise its stack trace will be lost.